crates/walrus-service/src/node/db_options.rs

```diff
@@ -0,0 +1,402 @@
+    /// Event store column family.
+    EventStore,
+    /// Init state column family.
+    InitState,
+
+    // Event blob writer column families.
+    /// Certified event blob column family.
+    Certified,
+    /// Attested event blob column family.
+    Attested,
+    /// Pending event blob column family.
+    Pending,
+    /// Failed to attest event blob column family.
+    FailedToAttest,
+
+    // Unknown/custom column family.
+    /// Custom column family with a given name.
+    Custom(String),
+}
+
+impl DbColumnFamily {
+    /// Create a DbColumnFamily from a column family name string.
+    pub fn from_name(name: &str) -> Self {
+        use crate::{
+            event::event_processor::db::constants as event_processor_constants,
+            node::{
+                event_blob_writer::{
+                    attested_cf_name,
+                    certified_cf_name,
+                    failed_to_attest_cf_name,
+                    pending_cf_name,
+                },
+                storage::constants::{
+                    aggregate_blob_info_cf_name,
+                    event_cursor_cf_name,
+                    metadata_cf_name,
+                    node_status_cf_name,
+                    per_object_blob_info_cf_name,
+                },
+            },
+        };
+
+        // Check for known column family names.
+        if name == node_status_cf_name() {
+            Self::NodeStatus
+        } else if name == metadata_cf_name() {
+            Self::Metadata
+        } else if name == aggregate_blob_info_cf_name() {
+            Self::AggregateBlobInfo
+        } else if name == per_object_blob_info_cf_name() {
+            Self::PerObjectBlobInfo
+        } else if name == event_cursor_cf_name() {
+            Self::EventCursor
+        } else if name == event_processor_constants::EVENT_STORE {
+            Self::EventStore
+        } else if name == event_processor_constants::INIT_STATE {
+            Self::InitState
+        } else if name == certified_cf_name() {
+            Self::Certified
+        } else if name == attested_cf_name() {
+            Self::Attested
+        } else if name == pending_cf_name() {
+            Self::Pending
+        } else if name == failed_to_attest_cf_name() {
+            Self::FailedToAttest
+        } else if let Some(shard_str) = name.strip_prefix("primary_sliver_") {
+            if let Ok(shard) = shard_str.parse::<u16>() {
+                Self::PrimarySliver(shard)
+            } else {
+                Self::Custom(name.to_string())
+            }
+        } else if let Some(shard_str) = name.strip_prefix("secondary_sliver_") {
+            if let Ok(shard) = shard_str.parse::<u16>() {
+                Self::SecondarySliver(shard)
+            } else {
+                Self::Custom(name.to_string())
+            }
+        } else if let Some(shard_str) = name.strip_prefix("shard_status_") {
+            if let Ok(shard) = shard_str.parse::<u16>() {
```

Comment:
Also, if we need to parse the shard index we can reuse `id_from_column_family_name` if we needed to.